x-pack/plugins/watcher/public/models/action/index.action.js

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+
+import { get } from 'lodash';
+import { BaseAction } from './base_action';
+import { i18n } from '@kbn/i18n';
+
+const requiredFields = ['host', 'port'];
+const optionalFields = [
+  'scheme',
+  'path',
+  'method',
+  'headers',
+  'params',
+  'auth',
+  'body',
+  'proxy',
+  'connection_timeout',
+  'read_timeout',
+  'url'
+];
+
+const allFields = [...requiredFields, ...optionalFields];
+
+export class IndexAction extends BaseAction {
+  constructor(props = {}) {
+    super(props);
+
+    this.fields = {};
+    allFields.forEach((field) => {
+      this.fields[field] = get(props, field);
+    });
+  }
+
+  get upstreamJson() {
+    // Add all required fields to the request body
+    let result = requiredFields.reduce((acc, field) => {
+      acc[field] = this.fields[field];
+      return acc;
+    }, super.upstreamJson);
+
+    // If optional fields have been set, add them to the body
+    result = optionalFields.reduce((acc, field) => {
+      if (this[field]) {
+        acc[field] = this.fields[field];
+      }
+      return acc;
+    }, result);
+
+    return result;
+  }
+
+  get description() {
+    return i18n.translate('xpack.watcher.models.indexAction.description', {
+      defaultMessage: 'The {index} will be indexed as {docType}',
+      values: {
+        index: this.index,
+        docType: this.doc_type,
```

Comment:
I think we need to update the fields references here:

* `this.index` -> `this.fields.index`
* `this.doc_type` -> `this.fields.doc_type`